delivery_roulier/tests/test_delivery_roulier.py

```diff
@@ -0,0 +1,117 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from unittest.mock import MagicMock, patch
+
+from odoo_test_helper import FakeModelLoader
+from roulier import roulier
+
+from odoo.tests.common import TransactionCase
+
+roulier_ret = {
+    "parcels": [
+        {
+            "reference": "",
+            "tracking": {"url": "", "number": "Test tracking"},
+            "label": {
+                "name": "label_test",
+                "data": b"dGVzdCBsYWJlbA==",
+                "type": "zpl2",
+            },
+            "id": 1,
+        }
+    ],
+    "annexes": [{"name": "annexe name", "type": "txt", "data": b"dGVzdCBhbm5leGU="}],
+}
+
+
+class DeliveryRoulierCase(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.loader = FakeModelLoader(cls.env, cls.__module__)
+        cls.loader.backup_registry()
+
+        # The fake class is imported here !! After the backup_registry
+        from .models import FakeDeliveryCarrier
+
+        cls.loader.update_registry((FakeDeliveryCarrier,))
+        cls.real_get_carriers_action_available = roulier.get_carriers_action_available
+
+    def setUp(self):
+        super().setUp()
+        delivery_product = self.env["product.product"].create(
```

Comment:
Why re-creating this static data for every test? Also can you disable tracking or use BaseCommon as base class?